internal/core/graphics/image.rs

```diff
@@ -360,6 +372,65 @@ impl ImageInner {
                     }
                 }
             }
+            ImageInner::StaticTextures(ts) => {
+                let mut buffer =
+                    SharedPixelBuffer::<Rgba8Pixel>::new(ts.size.width, ts.size.height);
+                let stride = buffer.stride() as usize;
+                let slice = buffer.make_mut_slice();
+                for t in ts.textures.iter() {
+                    let rect = t.rect.to_usize();
+                    for y in 0..rect.height() {
+                        let slice = &mut slice[(rect.min_y() + y) * stride..][rect.x_range()];
+                        let source = &ts.data[t.index + y * rect.width() * t.format.bpp()..];
+                        match t.format {
+                            PixelFormat::Rgb => {
+                                let mut iter = source.chunks_exact(3).map(|p| Rgba8Pixel {
+                                    r: p[0],
+                                    g: p[1],
+                                    b: p[2],
+                                    a: 255,
+                                });
+                                slice.fill_with(|| iter.next().unwrap());
+                            }
+                            PixelFormat::Rgba => {
+                                let mut iter = source.chunks_exact(4).map(|p| Rgba8Pixel {
+                                    r: p[0],
+                                    g: p[1],
+                                    b: p[2],
+                                    a: p[3],
+                                });
+                                slice.fill_with(|| iter.next().unwrap());
+                            }
+                            PixelFormat::RgbaPremultiplied => {
+                                let mut iter = source.chunks_exact(4).map(|p| {
+                                    let a = p[3];
+                                    Rgba8Pixel {
+                                        r: (p[0] as u16 * a as u16 / 255) as u8,
+                                        g: (p[1] as u16 * a as u16 / 255) as u8,
+                                        b: (p[2] as u16 * a as u16 / 255) as u8,
+                                        a,
+                                    }
+                                });
+                                slice.fill_with(|| iter.next().unwrap());
+                            }
```

Comment:
It seems to me that the RgbaPremultipiled and Rgba cases are swapped, since later `Some(SharedImageBuffer::RGBA8Premultiplied(buffer))` is returned.

```suggestion
                            PixelFormat::RgbaPremultiplied => {
                                let mut iter = source.chunks_exact(4).map(|p| Rgba8Pixel {
                                    r: p[0],
                                    g: p[1],
                                    b: p[2],
                                    a: p[3],
                                });
                                slice.fill_with(|| iter.next().unwrap());
                            }
                            PixelFormat::Rgba => {
                                let mut iter = source.chunks_exact(4).map(|p| {
                                    let a = p[3];
                                    Rgba8Pixel {
                                        r: (p[0] as u16 * a as u16 / 255) as u8,
                                        g: (p[1] as u16 * a as u16 / 255) as u8,
                                        b: (p[2] as u16 * a as u16 / 255) as u8,
                                        a,
                                    }
                                });
                                slice.fill_with(|| iter.next().unwrap());
                            }
```

Reply:
Nice catch